src/hooks/useDataSource/useDataSource.tsx

```diff
@@ -1,49 +1,41 @@
 import * as React from "react";
 
-type Refreshable = { refresh: () => void };
-
 type DataSuccess<T> = {
   state: "success";
   data: T;
-} & Refreshable;
+};
 
 type DataError = {
   state: "error";
-} & Refreshable;
+};
 
 type DataLoading = {
   state: "loading";
-} & Refreshable;
+};
+
+type Refreshable = {
+  refresh: () => void;
+};
 
-export type DataResult<T> = DataSuccess<T> | DataError | DataLoading;
+type Data<T> = DataSuccess<T> | DataError | DataLoading;
+export type DataResult<T> = Data<T> & Refreshable;
```

Comment:
Not a super important comment, but between "Data" and "DataResult" I wonder if the naming of these things can be made clearer.

Reply:
I've dropped Data and removed Refreshable from DataResult. Given the storybook thing, I've just explicitly had the hook return `DataResult & Refreshable` (storybook doesn't care about the refreshable bit in the type by the looks of it).

Reply:
Debatable at this point whether Refreshable needs to exist vs just being an explicit interface on the return value (`DataResult & { refresh: ... }`). Do you have a preference either way, @fildon?